lib/Gateway.js

```diff
@@ -502,15 +502,18 @@ Gateway.prototype.parsePayload = function (payload, valueId, valueConf) {
 
       // map modes coming from hass
       if (valueId.type === 'list' && valueId.values.indexOf(payload) < 0) {
-        var hassDevice = this.discovered[valueId.value_id]
-        payload = hassDevice.mode_map ? hassDevice.mode_map[payload] : payload
+        let hassDevice = this.discovered[valueId.value_id]
+        // for thermostat_fan_mode command class use the fan_mode_map
+        if (valueId.class_id === 0x44 && hassDevice.fan_mode_map) payload = hassDevice.fan_mode_map[payload]
```

Comment:
You should add a check that `hassDevice` is not defined

Reply:
Good idea. I added a check. See if that's good or if we need to specifically check for undefined.